_extras/for_instructors.md

```diff
@@ -0,0 +1,312 @@
+~~~
+sudo mkdir -p /etc/skel/.zsh
+sudo cp -r "$HOME/.zsh/pure" /etc/skel/.zsh/
+sudo cp "$HOME/.zshrc" /etc/skel/
+ 
+sudo cp "$HOME/.gitconfig" /etc/skel/ 
+
+sudo mkdir -p /etc/skel/.jupyter
+sudo cp "$HOME/.jupyter/jupyter_notebook_config.py" /etc/skel/.jupyter/
+~~~
+
+Now, at long last, you can create a new user from the JupyterHub admin panel.
+Using this user, login from another device or using the browser's incognito mode,
+and then check to see that everything functions as expected.
+
+### 3.8. Finally, add users!
+
+Depending on group size and logistics, we suggest creating users beforehand via the admin control panel (e.g. using their email addresses, or the username part of these), and letting users create their own passwords once they navigate to the hub URL for the first time. This is the default configuration.
+
+Different authentication options are possible (e.g. admin can also authenticate users individually after they sign up, or authentication can happen via Google, GitHub or AWS). Refer to the documentation for further options.
+
+
+## 4. Some notes and further options
+
+### EC2 instance notes
+
+- The letter `a` in type name (eg. `t3a.medium`) means that they are running on AMD, and are slightly cheaper than their Intel counterparts (without the `a`).
+- There are different instance types. We're interested in T3 and M5, both of which are multipurpose. T3 instances are burstable (meaning they accumulate credits when idling and use them when active, so don't offer full performance all the time, which suits us well). M5 are fixed-performance multipurpose (you get the full capacity all the time).
+- TLJH recommends `t3.small` to start with: 2 vCPU and 2 GB RAM should be enough to install everything (though some conda installs may be pushing the 2GB RAM limit for some reason).
+- We went with `t3a.medium` from the get-go: 2 vCPU and 4GB RAM are very comfortable for all setup steps, and it should be fine for a few concurrent users if you want to leave the hub running between workshop days - the price around $1 a day seems fine for that.
+- For the workshop sessions, consider upscaling to `M5a.4xlarge` (for the live sessions only, meaning downscaling back to `t3a.medium` overnight in a 2-day format). The "quadruple extra large" offers 16 vCPU and 64 GB RAM. Judging by looking at htop, this is serious overprovisioning even for ca 16 people, but at $1 an hour it seems a reasonable price for comfort.
+- To scale (up or down), go to EC2 management console, select your instance, right click (or click on Actions), and stop it. Then again right-click (or Actions), instance settings, change instance type. Then bring it back up.
+- Elastic IP incurs cost when not associated with a running instance, but it is marginal.
+- After the workshop, you will want to pull everything down, including storage (if left sitting, storage may generate a couple dollars per month).
+
```

Comment:
How about a little troubleshooting section? The idea to look at system log comes from TLJH docs, the things to look for I found myself

```suggestion

### Troubleshooting

- When checking if your hub was initialized successfully, apart from trying to access the server's ip address, you may want to view the AWS's system log.
- In EC2 Management Console, Instances page, right click your instance, and select "Monitor and troubleshoot", "Get system log".
- If system setup was finished, there will be a lot of text, most likely ending with something like:
   ~~~
   2022/12/01 12:40:30Z: OsProductName: Ubuntu
   2022/12/01 12:40:30Z: OsVersion: 22.04
   ~~~
- If the JupyterHub bootsrap script succeeded, within the last 30 lines you will find:
   ~~~
   [  210.143720] cloud-init[1233]: Waiting for JupyterHub to come up (1/20 tries)
   [  210.147437] cloud-init[1233]: Done!
   ~~~
- If the "user data" was not given, or pasted incorrectly (e.g. without admin name), within the last 30 lines you will probably see:
   ~~~
   cloud-init[1246]: 2022-12-01 12:40:29,790 - util.py[WARNING]: Running module scripts-user (<module 'cloudinit.config.cc_scripts_user' from '/usr/lib/python3/dist-packages/cloudinit/config/cc_scripts_user.py'>) failed
   ~~~
- In case of problems, you can log in through `ssh` and try running the TLJH bootstrap script manually, or terminate the instance and start over.
```